lang/py/src/avro/io.py

```diff
@@ -280,6 +298,83 @@ def read_utf8(self):
     """
     return unicode(self.read_bytes(), "utf-8")
 
+  def read_date_from_int(self):
+    """
+    Decode python date object from int
+    int stores the number of days from
+    the unix epoch, 1 January 1970 (ISO calendar).
+    """
+    days_since_epoc = self.read_int()
+    if days_since_epoc < 0:
```

Comment:
I think the negative number is still an valid value, which represents the date before 1/1/1970.